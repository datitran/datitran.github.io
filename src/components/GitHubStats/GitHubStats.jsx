import React from 'react';
import './GitHubStats.scss';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCode, faCodeBranch, faStar, faUsers } from '@fortawesome/free-solid-svg-icons'


export default (props) => {

    function openRepo() {
        console.log('Track: goToRepo');

        window.open(props.stats.html_url, '_blank');
    }

    return (
        <div className='gitHubStats__container' onClick={openRepo.bind(this)}>
            <div className='gitHubStats__stat'>
                <p className='gitHubStats__icon'><FontAwesomeIcon icon={faCode} /></p>
                <p className='gitHubStats__description'>{props.language}</p>
            </div>
            <div className='gitHubStats__stat'>
                <p className='gitHubStats__icon'><FontAwesomeIcon icon={faCodeBranch} /></p>
                <p className='gitHubStats__description'>{props.stats.forks_count}</p>
            </div>
            <div className='gitHubStats__stat'>
                <p className='gitHubStats__icon'><FontAwesomeIcon icon={faStar} /></p>
                <p className='gitHubStats__description'>{props.stats.stargazers_count}</p>
            </div>
            <div className='gitHubStats__stat'>
                <p className='gitHubStats__icon'><FontAwesomeIcon icon={faUsers} /></p>
                <p className='gitHubStats__description'>{props.stats.watchers_count}</p>
            </div>
        </div>
    );
}
