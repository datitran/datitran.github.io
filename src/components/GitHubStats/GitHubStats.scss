@import '../../styles/variables.scss';

.gitHubStats {
    &__container {
        display: flex;
        flex-direction: row;
        flex: 1;
        background-color: $lightestgray;
        padding: 10px;
        cursor: pointer;

        &:hover {
            > * > * {
                color: darken($darkestgray, 10%);
            }
        }
    }

    &__stat {
        flex: 1;
        display: flex;
        flex-direction: column;
        align-items: center;
        text-align: center;
    }

    &__icon {
        color: $darkestgray;
        font-size: 16px;
    }

    &__description {
        color: $darkgray;
        font-size: 14px;
        margin-top: 10px;
    }
}