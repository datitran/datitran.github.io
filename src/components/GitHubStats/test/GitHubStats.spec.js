import React from 'react';
import renderer from 'react-test-renderer';
import GitHubStats from '../GitHubStats';

describe('GitHubStats component', () => {
    it('should render as expected', () => {
        const renderedComponent = renderer.create(
            <GitHubStats stats={{}} />
        );
        expect(renderedComponent.toJSON()).toMatchSnapshot();
    })
});