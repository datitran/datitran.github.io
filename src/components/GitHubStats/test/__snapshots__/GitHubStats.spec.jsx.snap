// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubStats component should render as expected 1`] = `
<div
  className="gitHubStats__container"
  onClick={[Function]}
>
  <div
    className="gitHubStats__stat"
  >
    <p
      className="gitHubStats__icon"
    >
      <span
        aria-hidden={true}
        className="fa fa-code"
      />
    </p>
    <p
      className="gitHubStats__description"
    />
  </div>
  <div
    className="gitHubStats__stat"
  >
    <p
      className="gitHubStats__icon"
    >
      <span
        aria-hidden={true}
        className="fa fa-code-fork"
      />
    </p>
    <p
      className="gitHubStats__description"
    />
  </div>
  <div
    className="gitHubStats__stat"
  >
    <p
      className="gitHubStats__icon"
    >
      <span
        aria-hidden={true}
        className="fa fa-star"
      />
    </p>
    <p
      className="gitHubStats__description"
    />
  </div>
  <div
    className="gitHubStats__stat"
  >
    <p
      className="gitHubStats__icon"
    >
      <span
        aria-hidden={true}
        className="fa fa-users"
      />
    </p>
    <p
      className="gitHubStats__description"
    />
  </div>
</div>
`;
